src/runtime/methodbinder.cs

```diff
@@ -127,9 +176,14 @@ internal static MethodInfo MatchSignatureAndParameters(MethodInfo[] mi, Type[] g
                 {
                     continue;
                 }
+
                 for (var n = 0; n < pi.Length; n++)
                 {
-                    if (sigTp[n] != pi[n].ParameterType)
+                    Type sig = sigTp[n];
+                    Type param = pi[n].ParameterType;
+
+                    if (!param.IsGenericParameter && !IsNullableOf(sig, param) &&
+                        !param.IsAssignableFrom(sig))
```

Comment:
Can you explain what this method was supposed to be doing, and what your change enables it to do now? It is actually not clear what genericTp and sigTp represent from their names :(